woocommerce/woocommerce
# Add introduction banner card into multichannel marketing page

### All Submissions:


### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/34904. Previous draft PR is PR https://github.com/woocommerce/woocommerce/pull/36220.

In this PR, we add the introduction banner card at the top of the multichannel marketing page.

There are two "types" of introduction banner card: one with the "Create a campaign" and "Add channels" buttons, and one without.

Screenshot of introduction banner card without the "Create a campaign" and "Add channels" buttons:

![image](https://user-images.githubusercontent.com/417342/209806777-a220e7f2-8937-45b2-bc07-726a2d0a2b19.png)

Screenshot of introduction banner card with the "Create a campaign" and "Add channels" buttons:

![image](https://user-images.githubusercontent.com/417342/209806644-dad37194-5224-4f60-98b0-353c9d1eb7a8.png)

Users can dismiss the card by clicking at the top right "X" button. We store the dismissed state by using a WP Option `woocommerce_marketing_overview_multichannel_banner_dismissed`. This approach is similar to the existing [`woocommerce_marketing_overview_welcome_hidden`](https://github.com/woocommerce/woocommerce/blob/3daeac9a698f71f84619a37335d5289598a5b394/plugins/woocommerce-admin/client/marketing/overview/welcome-card/index.js#L21-L23) WP Option for the WelcomeCard in the existing Marketing page.

The "Creata a campaign" button is displayed when there is at least one registered channel. Clicking it will display the "Create a campaign" modal.

The "Add channels" button is displayed when there is at least one registered channel and one recommended channel. When users click on the "Add channels" button, the page will scroll down to the "Add channels" button in the Channels card, allowing users to install a new channel.

### How to test the changes in this Pull Request:


Pre-conditions: 

- Enable WooCommerce Multichannel Marketing experience in WooCommerce Settings > Advanced > Features page: `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
- If you have dismissed the banner card, you can make it show up again by deleting the WP Option `woocommerce_marketing_overview_multichannel_banner_dismissed` in the database.

1. Go to WooCommerce Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`.
2. You should see the introduction banner card at the top of the page.
3. Proceed to setup a channel, and then come back to this marketing page.
4. Click on the "Create campaign" button. You should see the "Create a campaign" modal.
5. Click on the "Add channels" button. The page should scroll down to the "Add channels" button in the Channels card.
6. Dismiss the card, and reload the page. The card should not be displayed.



### Other information:


### FOR PR REVIEWER ONLY:
